src/UserSpaceInstrumentation/InstrumentProcess.cpp

```diff
@@ -102,6 +111,93 @@ bool IsBlocklisted(std::string_view function_name) {
+// the clone syscall. This thread is used to execute the initialization code inside the target.
+// Note that calling the result of the clone call a "thread" is a bit of a misnoumer: We
+// do not create a new data structure for thread local storage but use the one of the thread we
+// halted.
+ErrorMessageOr<MachineCode> MachineCodeForCloneCall(pid_t pid, void* library_handle,
+                                                    uint64_t top_of_stack) {
+  constexpr uint64_t kCloneFlags =
+      CLONE_FILES | CLONE_FS | CLONE_IO | CLONE_SIGHAND | CLONE_SYSVSEM | CLONE_THREAD | CLONE_VM;
+  constexpr const char* kInitializeInstrumentationFunctionName = "InitializeInstrumentation";
+  OUTCOME_TRY(auto&& initialize_instrumentation_function_address,
+              DlsymInTracee(pid, library_handle, kInitializeInstrumentationFunctionName));
+  MachineCode code;
+  code.AppendBytes({0x48, 0xbf})
+      .AppendImmediate64(kCloneFlags)  // mov kCloneFlags, %rdi
+      .AppendBytes({0x48, 0xbe})
+      .AppendImmediate64(top_of_stack)  // mov thread_stack_memory, %rsi
+      .AppendBytes({0x48, 0xba})
+      .AppendImmediate64(0x0)  // mov parent_tid, %rdx
+      .AppendBytes({0x49, 0xba})
+      .AppendImmediate64(0x0)  // mov child_tid, %r10
+      .AppendBytes({0x49, 0xb8})
+      .AppendImmediate64(0x0)                                   // mov tls, %r8
+      .AppendBytes({0x48, 0xc7, 0xc0, 0x38, 0x00, 0x00, 0x00})  // mov 0x38, rax
+      .AppendBytes({0x0f, 0x05})                                // syscall (clone)
+      .AppendBytes({0x48, 0x85, 0xc0})                          // testq	%rax,%rax
+      .AppendBytes({0x0f, 0x84, 0x01, 0x00, 0x00, 0x00})        // jz 0x01(rip)
+      .AppendBytes({0xcc})                                      // int3
+      .AppendBytes({0x48, 0xb8})
+      .AppendImmediate64(absl::bit_cast<uint64_t>(
+          initialize_instrumentation_function_address))  // mov initialize_instrumentation, rax
+      .AppendBytes({0xff, 0xd0})                         // call rax
+      .AppendBytes({0x48, 0xc7, 0xc7, 0x00, 0x00, 0x00, 0x00})  // mov 0x0, rdi
+      .AppendBytes({0x48, 0xc7, 0xc0, 0x3c, 0x00, 0x00, 0x00})  // mov 0x3c, rax
+      .AppendBytes({0x0f, 0x05});                               // syscall (exit)
+  return code;
+}
+
+void WaitForThreadToExit(pid_t pid, pid_t tid) {
+  auto tids = orbit_base::GetTidsOfProcess(pid);
+  while (std::find(tids.begin(), tids.end(), tid) != tids.end()) {
+    std::this_thread::sleep_for(std::chrono::milliseconds{1});
+    tids = orbit_base::GetTidsOfProcess(pid);
+  }
+}
+
+ErrorMessageOr<std::vector<pid_t>> GetNewOrbitThreads(
+    pid_t pid, const absl::flat_hash_set<pid_t>& tids_before_injection) {
+  constexpr int kNumberOfRetries = 3;
+  static const absl::flat_hash_set<std::string> kValidOrbitThreadNames{
+      "default-executo", "resolver-execut", "grpc_global_tim", "ConnectRcvCmds", "ForwarderThread"};
```

Comment:
There are 5 names here, this implies that we are spawning two threads with the same name since `kNumberOfThreadsToSpawn == 6`, right? Maybe we should check that it is the expected thread that is spawned twice?